mintpy/objects/cluster.py

```diff
@@ -1,102 +1,276 @@
-        print(msg)
-        config_name = cluster_type
-
-    return config_name
-
-
-def format_walltime(walltime, cluster_type):
-    """format the walltime str for different clusters
-    HH:MM:SS - pbs / slurm
-    HH:MM    - lsf
-    """
-    num_digit = len(walltime)
-    if cluster_type in ['pbs', 'slurm'] and num_digit != 8:
-        if num_digit == 5:
-            walltime += ":00"
+import glob
+import os
+import shutil
+import time
+import dask
+from dask.distributed import LocalCluster, Client, as_completed
+
+
+class DaskCluster:
+
+    def __init__(self, cluster_type, num_workers, write_job_script=True, **kwargs):
+        """Generic dask cluster wrapper"""
+
+        self.cluster = None
+        self.client = None
+        self.num_workers = num_workers
+
+        # Properly format cluster type for consistency
+        cluster_type = cluster_type.lower()
+        cluster_list = ['lsf', 'pbs', 'slurm', 'local']
+        if cluster_type not in cluster_list:
+            msg = "Cluster type '{}' not supported".format(cluster_type)
+            msg += '\nSupported cluster types: {}'.format(cluster_list)
+            raise ValueError(msg)
+        print("Dask cluster type: {}".format(cluster_type))
+
+
+        # for local cluster, NO need to do the extra configuration
+        if cluster_type == 'local':
+            self.cluster = LocalCluster()
         else:
-            raise ValueError("input walltime ({}) not in HH:MM:SS or HH:MM format.".format(walltime))
 
-    elif cluster_type in ['lsf'] and num_digit != 5:
-        if num_digit == 8:
-            walltime = walltime[:5]
+            # import related dask module only
+            # because job_queue is not available on macports, which make sense
+            try:
+                import dask_jobqueue as jobqueue
+            except ImportError:
+                raise ImportError('Cannot import dask_jobqueue!')
+
+            # check input config name
+            if 'config_name' in kwargs.keys():
+                kwargs['config_name'] = self.format_config_name(kwargs['config_name'], cluster_type)
+            print("Dask config name: {}".format(kwargs['config_name']))
+
+            # check walltime format for each cluster type
+            if 'walltime' in kwargs.keys():
+                kwargs['walltime'] = self.format_walltime(kwargs["walltime"], cluster_type)
+            print('Dask worker walltime: {}'.format(kwargs['walltime']))
+
+            # initiate cluster object
+            if cluster_type == 'lsf':
+                self.cluster = jobqueue.LSFCluster(**kwargs)
+            elif cluster_type == 'pbs':
+                self.cluster = jobqueue.PBSCluster(**kwargs)
+            elif cluster_type == 'slurm':
+                self.cluster = jobqueue.SLURMCluster(**kwargs)
+
+            if write_job_script:
+                self.write_job_script()
+
+        self.cluster.scale(num_workers)
```

Comment:
So the variable `NUM_WORKERS` is not required? I thought @2gotgrossman put it there for a reason.

Reply:
NUM_WORKERS was just a wrapper variables around inps.numWorker as far as I could tell.

Reply:
Looks good to me, as long as it works!

Reply:
I used this PR all day today with no issues, so it sees fine.